base/sort.jl

```diff
@@ -426,7 +426,7 @@ function selectperm!{I<:Integer}(ix::AbstractVector{I}, v::AbstractVector,
                                  order::Ordering=Forward,
                                  initialized::Bool=false)
     if !initialized
-        @inbounds for i = 1:length(ix)
+        @inbounds for i in eachindex(ix)
```

Comment:
Didn't know about that. Do you have specific type of permutation in mind? 

I'll create a PR to fix this.